### ISSUE TITLE
Message draft from content shared into Signal reappears after being sent and resuming app window

### ISSUE THREAD

**Issue by wesinator** [signalapp/Signal-Android#8947]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When content is shared into signal and sent, resuming the Signal window does not reset the message draft or return to the main messages list
see also #4729

### Steps to reproduce
- In other app, "Share" content to Signal, existing convo
- send message
- switch out of signal app, then back 

**Actual result:** shared text that was already sent is still populated in draft message after resuming the app . Requires clearing out the draft text, closing Signal and reopening to reset the message convo thread. 
 
Also, Signal does not return to the messages list after shared text - #8074

**Expected result:** Clear the shared content draft after sending, and return to main message list view after sharing.

### Device info

**Device:** Nokia 5
**Android version:** 9
**Signal version:** 4.43.8

### Link to debug log

https://debuglogs.org/6c595ee8289e9f90321b4a6e4b6e01e3a4d7cccb486d407c4f39697970e5ab1e

**Comment by philipreimer** [comment]
I also experience this bug most of the times when I share something to Signal. But not always, and like @wwwizzarrdry could not find a consistent pattern to reproduce this bug.

Usually, it happens if I do  the following:

1. Tap share button in some application (for me mostly Firefox)
1. Tap Signal
1. Select existing conversation from Signal conversation list
1. Tap Send
1. Tap "back" button - that is Android's back button at the bottom of the screen, *not* Signal's back button in the top left corner
1. -> Previous app (Firefox) gets focus again
1. Open "App switcher" (or however the thing is called which allows you to switch between different running apps)
1. Select Signal

### Result
Signal shows draft message with shared content from above. I have to manually clear the input box.

Sometimes tapping the "back" button in step 5 will not return to Firefox immediately but to Signal's conversation list. I then have to tap the "back" button again to return to Firefox.
In those cases, it seems that the draft message does not appear again when returning to Signal in step 8. But I am not 100% certain that this is always the case.


```
Device: Nokia 8.1
Android: 10
Signal: 4.58.5
```

**Comment by wwwizzarrdry** [comment]
It is likely to be related to how Signal + Android OS manage the text associated with the "share" feature in Android.

Maybe signal has its own local storage associated with text copied via the native share function in Android?

For example, I'm posting to this thread on my Android phone and just long-pressed and selected paste (right here in the text field). It pasted a uri I had previously copied 2 days ago. So we know Android stores your clipboard (likely) indefinitely or until memory wipe on reboot. 

However, when I opened Signal just now, the uri was not immediately pre-pasted into the message box.

Also, I just tested sharing a website uri to signal.
1. Opened Brave to a website.
2. Selected Share from the brave menu.
3. Chose Signal as the app to share to.
4. Selected a recipient.
5. The text was pre-pasted as expected.
6. I did not send the message.
7. Returned to Brave (thereby closing the ephemeral instance of signal)
8. I returned to the home screen and opened Signal, then entered a conversation.
9. The copied uri text did not pre-populate.
---
*. I repeated these steps, but this time I sent the pasted message.
10. Reopened Signal, entered a conversation, and now, too, the copied text was not still pre-populated in the message box (as is the expected behavior).

Hmm, so did signal fix this bug? I could have sworn that I just experienced the "phantom" paste event very recently, but then again, it's kind of like the Shining around here and I may be losing my mind in quarantine...

**Comment by philipreimer** [comment]
@wwwizzarrdry In the case where I did *not* send the message I would expect that Signal keeps the draft message and I can access the draft message when I open Signal again. And at least for me that seems to work all the time.

**Comment by philipreimer** [comment]
Any update? The issue still exists in Signal 4.64.4

**Comment by philipreimer** [comment]
The issue still exists in Signal 4.71.5

Any chance of getting this fixed? It is a very annoying "feature".

**Comment by infinitewaveparticle** [comment]
@fumiakiy Will this be fixed anytime soon? It's been happening for 2 years.

**Comment by fumiakiy** [comment]
Well this issue could be worked around by merging the pull request. When and if it would be merged (if ever) is not up to me though.

Merging a PR is a strange and complicated decision; it's strange because merging a PR means the product owner accepts the responsibility of maintaining a feature (or a bug) that the code written by a stranger may add to their product. You never know if the original author of the code would keep being responsible for what could happen by adding the code to the product.

All I could do is to write good enough code, a good commit message and a good pull request description, so good that the devs might feel less burdens pulling in lines code a stranger wrote for their product.

**Comment by infinitewaveparticle** [comment]
Makes sense. Signal seems to be painfully slow at fixing what they've messed up (bugs... Including extremely serious ones) - if they do it at all - and extremely focused on adding new features which only introduces more and more bugs that they'll never fix.... SMH... If this keeps up, then more and more people who actually care about what Signal claims to provide will be leaving the app.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by wwwizzarrdry** [comment]
This is still a recurring problem in version 5.28.10

**Comment by infinitewaveparticle** [comment]
I can also confirm this. Not fixed.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by Hanl0n** [comment]
This is still not fixed.
Here is a debuglog with two examples:
https://debuglogs.org/android/5.32.0/4f2e3632c05656bc203d34ef9e2f1813de5be121aad7ba86be350f6f6e536029

1644422496848 and 1644423850911 have this bug.

1644423894509 does not.

**Comment by alex-signal** [comment]
Skipping the intent if you re-open from recents sounds ok, but I'm wondering if it'll mess things up with search or other things like that. It's worth discussing so I brought it up with the other team members.

**Comment by infinitewaveparticle** [comment]
This is still very much an issue

Sent from ProtonMail mobile

-------- Original Message --------
